viewer/packages/camera-manager/src/CameraManager.ts

```diff
@@ -0,0 +1,457 @@
+/*!
+ * Copyright 2021 Cognite AS
+ */
+
+import * as THREE from 'three';
+import TWEEN from '@tweenjs/tween.js';
+import { ComboControls } from './ComboControls';
+import { CallbackData } from './types';
+
+export class CameraManager extends THREE.EventDispatcher {
+    public controls: ComboControls;
+    
+    private _camera: THREE.PerspectiveCamera;
+    private _domElement: HTMLElement;
+    
+    private modelRaycast: (x: number, y: number) => Promise<CallbackData>;
+    
+    private isDisposed = false;
+    public useOnClickTargetChange: boolean = false;
+    public automaticNearFarPlane: boolean = true;
+    private readonly _animationDuration: number = 600;
+    private readonly _minDefaultAnimationDuration: number = 600;
+    private readonly _maxDefaultAnimationDuration: number = 2500;
+    private readonly _minDistanceDefault: number = 0.1;
+    private readonly _useScrollTargetControls: boolean = true;
+    private readonly _automaticControlsSensitivity: boolean = false;
+    private readonly _canInterruptAnimations: boolean = false;
+    private readonly _raycaster: THREE.Raycaster = new THREE.Raycaster();
+
+    /**
+    * Reusable buffers used by functions in Cognite3dViewer to avoid allocations.
+    */
+    private readonly _updateNearAndFarPlaneBuffers = {
+        cameraPosition: new THREE.Vector3(),
+        cameraDirection: new THREE.Vector3(),
+        nearPlaneCoplanarPoint: new THREE.Vector3(),
+        nearPlane: new THREE.Plane(),
+        corners: new Array<THREE.Vector3>(
+            new THREE.Vector3(),
+            new THREE.Vector3(),
+            new THREE.Vector3(),
+            new THREE.Vector3(),
+            new THREE.Vector3(),
+            new THREE.Vector3(),
+            new THREE.Vector3(),
+            new THREE.Vector3()
+        )
+    };
+
+    /**
+   * Calculates new target when raycaster doesn't have any intersections with the model.
+   * @param cursorPosition Cursor position for desired calculations.
+   * @param cursorPosition.x
+   * @param cursorPosition.y
+   */
+  private calculateMissedRaycast = (cursorPosition: { x: number; y: number }, modelsBB: THREE.Box3): THREE.Vector3 => {
```

Comment:
```suggestion
  private calculateMissedRaycast(cursorPosition: { x: number; y: number }, modelsBB: THREE.Box3): THREE.Vector3  {
```
Why usually use this way of defining functions. No biggie, but nice to standardise (if there aren't any good reasons why we should do something else)

Reply:
Changed to normal